google/auth/credentials_async.py

```diff
@@ -75,42 +74,11 @@ class AnonymousCredentials(credentials.AnonymousCredentials, Credentials):
     """Credentials that do not provide any authentication information.
 
     These are useful in the case of services that support anonymous access or
-    local service emulators that do not use credentials.
+    local service emulators that do not use credentials. This class inherits
+    from the sync anonymous credentials file, but is kept if async credentials
+    is initialized and we would like anonymous credentials.
     """
 
-    '''
-    @property
-    def expired(self):
-        """Returns `False`, anonymous credentials never expire."""
-        return False
-
-    @property
-    def valid(self):
-        """Returns `True`, anonymous credentials are always valid."""
-        return True
-
-    def refresh(self, request):
-        """Raises :class:`ValueError``, anonymous credentials cannot be
-        refreshed."""
-        raise ValueError("Anonymous credentials cannot be refreshed.")
-
-    def apply(self, headers, token=None):
-        """Anonymous credentials do nothing to the request.
-
-        The optional ``token`` argument is not supported.
-
-        Raises:
-            ValueError: If a token was specified.
-        """
-        if token is not None:
-            raise ValueError("Anonymous credentials don't support tokens.")
-
-    def before_request(self, request, method, url, headers):
-        """Anonymous credentials do nothing to the request."""
-
-    def with_quota_project(self, quota_project_id):
-        raise ValueError("Anonymous credentials don't support quota project.")
-    '''
 
 @six.add_metaclass(abc.ABCMeta)
 class ReadOnlyScoped(credentials.ReadOnlyScoped):
```

Comment:
I notice that this doesn't override anything from credentials.ReadOnlyScoped. Is there a reason for having an inheritance here rather than just aliasing credentials.ReadOnlyScoped? (Probably fine to have it like this also).

Reply:
The reason I did it like this was that if we were to define an async credentials (service_account_async credentials for example), we would inherit from this file. In that case, I wanted a separate thread of all the classes through-out so that we then have an inheritance in service_account_async of credentials_async.ReadOnlyScoped instead of credentials.ReadOnlyScoped. However, since all the methods in the async version are synchronous, we do not require any added functionality in this class which is is why it is defined but left as is.

Reply:
I am currently drawing out the dependency graph and will add it to the design doc by tonight

Reply:
Thanks, I saw this in your design doc-- definitely helps!